nodebuilder/header/service.go

```diff
@@ -51,6 +52,33 @@ func (s *Service) GetVerifiedRangeByHeight(
 }
 
 func (s *Service) GetByHeight(ctx context.Context, height uint64) (*header.ExtendedHeader, error) {
+	// TODO(vgonkivs): remove after https://github.com/celestiaorg/go-header/issues/32 will be
+	// implemented
+	head, err := s.store.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if uint64(head.Height()) < height {
+		return nil, fmt.Errorf("header: given height exceeds local head."+
+			"localHeadHeight:%d, requestedHeight:%d", head.Height(), height)
+	}
+	if uint64(head.Height()) == height {
+		return head, nil
+	}
+
+	head, err = s.syncer.Head(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if uint64(head.Height()) < height {
+		return nil, fmt.Errorf("header: given height exceeds network head."+
```

Comment:
```suggestion
		return nil, fmt.Errorf("header: given height is from the future"+
```

Reply:
Let's make it clear to the user what the error means. In the current version, it tells that the user gave wrong input, while it's not.